lib/test/Test/Pos/Ssc/Toss/BaseSpec.hs

```diff
@@ -616,6 +622,33 @@ checksBadCertsPayload (GoodPayload epoch sgs certsMap mrs) pk cert =
         not (vcVssKey cert `elem` allVssKeys))
        ==> res1 .&&. res2 .&&. res3 .&&. res4
 
+-- [CT-99]: Test that no two stakeholders can have same active VSS keys
+checksBijectiveCertsPayload :: HasConfiguration => GoodCertsPayload -> VssCertData -> Property
+checksBijectiveCertsPayload (GoodPayload _ _ vssCertsMap _) _ = do
+    -- 1. A VSS payload with two certs that have equal VSS keys should not be
+    -- deserializable (with decode).
+
+    -- insert same certificate with different ids
+    let certsMap = getVssCertificatesMap vssCertsMap
+    let x@(_, cert1) : (sid2, _) : rest = HM.toList certsMap
+    let newCerts = UnsafeVssCertificatesMap $ HM.fromList $ x : (sid2, cert1) : rest
+
+    -- check that certificate duplication throws error on decoding stage
+    let encodedCerts = encode newCerts
+    let decodedCerts = deserialiseFromBytes (decode @VssCertificatesMap)
+                                            (toLazyByteString encodedCerts)
+    let check1 = HM.size certsMap >= 2 ==> isLeft decodedCerts
+
+    -- 2. A VSS payload should not be allowed to contain a certificate with a
+    -- VSS key that already exists in VssCertData. It looks like something
+    -- similar is checked in Test.Pos.Ssc.Toss.BaseSpec, but I'm not quite sure.
+    -- It's definitely possible to write a simpler, foolproof test, though.
+    -- TODO: 'checksBadCertsPayload' doesn't have 'VssCertData' in its arguments
```

Comment:
`checksBadCertsPayload` has `GoodCertsPayload` argument which has `SscGlobalState` inside and it in turn has `VssCertData` (`sgs ^. sgsVssCertificates`).
See `res4` in `checksBadCertsPayload`, it's very similar to what you need to write.

Reply:
Btw, what is _VSS Payload_? Is it _GoodPayload_ or something else?

Reply:
There is `SscPayload` type:
```
data SscPayload
    = CommitmentsPayload
        { spComms :: !CommitmentsMap
        , spVss   :: !VssCertificatesMap }
    | OpeningsPayload
        { spOpenings :: !OpeningsMap
        , spVss      :: !VssCertificatesMap }
    | SharesPayload
        { spShares :: !SharesMap
        , spVss    :: !VssCertificatesMap }
    | CertificatesPayload
        { spVss    :: !VssCertificatesMap }
    deriving (Eq, Show, Generic)
```

As you can see, all its constructors contain `VssCertificates`. I guess that's what is called `VSS Payload` there.

`GoodPayload` is just some ad-hoc type for tests.